CICD.Tools.SDKChecker/Program.cs

```diff
@@ -47,73 +46,53 @@ public static async Task<int> Main(string[] args)
-        /// Retrieves all projects not using SDK style.
-        /// </summary>
-        /// <param name="pathToSolution">Directory containing the .sln</param>
-        /// <returns>A collection of project names.</returns>
-        private static ISet<string> RetrieveLegacyStyleProjects(string pathToSolution)
+        
+        private static async Task Process(string workspace, string repoName, string branch, string solutionFilepath)
         {
-            if (!MSBuildLocator.IsRegistered)
-            {
-                MSBuildLocator.RegisterDefaults();
-            }
+            DevOpsMetrics metrics = new DevOpsMetrics();
 
-            HashSet<string> projectsWithLegacyStyle = new HashSet<string>();
-            foreach (var projectFile in GetProjects(pathToSolution))
+            if (String.IsNullOrWhiteSpace(solutionFilepath))
             {
-                var projectFileProcessor = new ProjectFile(projectFile);
-                if (!projectFileProcessor.UsesSDKStyle())
-                {
-                    projectsWithLegacyStyle.Add(Path.GetFileNameWithoutExtension(projectFile));
-                }
+                solutionFilepath = FileSystem.Instance.Directory.EnumerateFiles(workspace, "*.sln", SearchOption.AllDirectories).FirstOrDefault();
             }
-
-            return projectsWithLegacyStyle;
-        }
-
-        private static IList<string> GetProjects(string solutionFilePath)
-        {
-            List<string> projects = new();
-
-            var workspace = MSBuildWorkspace.Create();
-
-            var solution = Task.Run(() => workspace.OpenSolutionAsync(solutionFilePath)).GetAwaiter().GetResult();
-
-            foreach (var project in solution.Projects)
+            
+            if (String.IsNullOrWhiteSpace(solutionFilepath))
             {
-                projects.Add(project.FilePath);
+                throw new InvalidOperationException("Could not locate a solution file (.sln) in workspace: " + workspace);
             }
 
-            return projects;
-        }
+            Solution solution = Solution.Load(solutionFilepath);
 
-        private static async Task Process(string workspace, string repoName, string branch)
-        {
-            DevOpsMetrics metrics = new DevOpsMetrics();
-            var pathToSolution = FileSystem.Instance.Directory.EnumerateFiles(workspace, "*.sln", SearchOption.AllDirectories).FirstOrDefault();
-            if (String.IsNullOrWhiteSpace(pathToSolution))
+            List<string> legacyStyleProjects = new List<string>();
+            foreach (var projectInSolution in solution.Projects)
             {
-                throw new InvalidOperationException("Could not located a solution file (.sln) in workspace: " + workspace);
-            }
+                var project = solution.LoadProject(projectInSolution);
 
-            var projectsWithPackageConfig = RetrieveLegacyStyleProjects(pathToSolution);
+                if (project is not { ProjectType: ProjectType.Sdk })
```

Comment:
If parsers was changed, this may need to become ProjectStyle

Reply:
Done